marcdomain/Fast-Food-Fast
# #160720030 Ft place order api endpoint

### What does this PR do?
- Implementation of POST orders endpoint
### Description of Task to be completed
- Create place-order controller
- Create router for place-order
- Validate user input
- Write test for this endpoint
- Update README to include the endpoint
### How should this be manually tested?
- Using Postman, visit https://marcus-fast-food-fast.herokuapp.com/api/v1/auth/login  to Login admin using email: _admin@gmail.com_  password: _adminuser_
- Paste the generated token in the header authorization field
- Post menu by triggering POST on https://marcus-fast-food-fast.herokuapp.com/api/v1/menu
- Signup user by triggering POST on https://marcus-fast-food-fast.herokuapp.com/api/v1/auth/signup and paste the token into the header authorization field
- Place order using postman to trigger POST on https://marcus-fast-food-fast.herokuapp.com/api/v1/orders
- Check README for the test coverage
### Any background context you want to provide?
- User has the option of supplying delivery location or leaving it empty/undefined.
- If location is empty/undefined, delivery location defaults to the user residential address.
- Else, the location supplied supersedes the user residential address
### What are the relevant pivotal tracker stories?
- [#160720030](https://www.pivotaltracker.com/story/show/160720030)
